qiskit/qasm/qasmlexer.py

```diff
@@ -124,7 +124,7 @@ def t_MATCHES(self, t):
         return t
 
     def t_STRING(self, t):
-        r"\"([^\\\"]|\\.)*\" "
+        r"\"([^\\\"]|\\.)*\""
```

Comment:
This _might_ be dangerous, but I don't 100% know - the Python version of `lex` interprets docstrings, so the whitespace might be significant. (But I very much don't know.)  That said, if this commit does have an effect, then I think your new form is correct anyway.

Reply:
Oh yeah, now I remember. `black` will insert that space again.